java/Anton-Kozyk/src/com/zoolatech/lecture2/tasks/_2/Task2.java

```diff
@@ -0,0 +1,189 @@
+ * give any coordinates and direction as the initial robot location. Robot must not be able
+ * to get outside of a room - in such case the robot should move as much as possible to the
+ * room border, print a warning message and stop. All next moves in the same direction should
+ * be prevented.
+ */
+
+/**
+ When you first run the program, you will be asked to enter the dimensions of the
+ room (width and height) - they cannot be negative or zero. Next you will be able to enter the initial
+ coordinates of the robot, to do this, enter "y" and then specify the desired values,
+ keep in mind that these values can only be within the previously defined room
+ dimensions. To continue with the default values of the coordinates of the robot,
+ simply type any character and press Enter. Now the robot will follow your instructions
+ from the code. He can go forward a certain number of steps, or turn left and right. Also
+ note that left or right is relative to the direction of the robot. That is, if the robot
+ is standing in the direction of the South, turning to the left, it will turn in the
+ direction of the East. If the number of steps entered exceeds the size of the room, the
+ robot will stop at its border. printPosition() method displays the compass, current
+ position, direction, as well as a map with a picture of a robot on it (it can be displayed
+ in 4 ways, depending on the direction - [ > ], [ < ], [/\ ] , [\/ ]).
+ */
+
+public class Task2 {
+    public static void main(String[] args) {
+        Robot robot = new Robot();
+        robot.printPosition();
+        robot.moveForward(10);
+        robot.turnLeft();
+        robot.moveForward(4);
+        robot.turnLeft();
+        robot.moveForward(3);
+        robot.printPosition();
+        robot.turnLeft();
+        robot.printPosition();
+        robot.moveForward(4);
+        robot.printPosition();
+    }
+}
+
+class Robot {
+    //Start position coordinates
+    int x = 0;
+    int y = 0;
+    char direction = 'S';
+
+    int W; //width of area
+    int H; //height of area
+
+    { //Initialization block
+        Scanner scanner = new Scanner(System.in);
+
+        do {
+            System.out.print("Enter Width of area: ");
+            W = scanner.nextInt();
+        } while (W <= 0);
+
+        do {
+            System.out.print("Enter Height of area: ");
+            H = scanner.nextInt();
+        } while (H <= 0);
+
+        System.out.print("If you want to chose start coordinates type \"y\" or" +
+                "type any letter to continue with default [0:0]: ");
+        if (scanner.next().charAt(0) == 'y') {
+            do {
+                System.out.print("Enter x: ");
+                x = scanner.nextInt();
+            } while (x < 0 || x >= W);
+
+            do {
+                System.out.print("Enter y: ");
+                y = scanner.nextInt();
+            } while (y < 0 || y >= H);
+        }
+    }
+
+    public void moveForward(int step) {
```

Comment:
> The class should provide methods to move a robot forward by either 1 or N tiles

It was meant to have two method - one that always moves a robot by 1 tile, second - that accepts number of tiles. Sorry for a confusion.